packages/mattermost-redux/src/reducers/entities/apps.ts

```diff
@@ -77,7 +91,28 @@ const rhs = combineReducers({
     forms: rhsForms,
 });
 
+export function pluginEnabled(state = true, action: GenericAction): boolean {
+    switch (action.type) {
+    case AppsTypes.APPS_PLUGIN_ENABLED: {
+        return true;
+    }
+    case AppsTypes.APPS_PLUGIN_DISABLED: {
+        return false;
+    }
+    case AppsTypes.RECEIVED_APP_BINDINGS: {
+        return true;
+    }
+    case AppsTypes.FAILED_TO_FETCH_APP_BINDINGS: {
+        return false;
```

Comment:
If the status is set to false, when do we expect to become true again? I understand one is on plugin enabled, but are we trying in any other place to see if the plugin is enabled?

Should we make a ping function for when the client gets "back online"? (Getting back from the background, recovering the internet connection...)

Reply:
@larkox there is a call to fetch bindings when the client reconnects after being offline (in this PR and the mobile PR). This doubles as the check to see if the plugin is enabled.

Maybe we should avoid the call to fetch bindings on reconnect and instead just ping the Apps plugin to see if it's enabled. This would avoid the situation where all users request bindings on reconnect simultaneously, e.g. when the server is restarted. cc @hanzei

Reply:
Regarding the second part of your comment, I sincerely don't know what direction to take. I feel that just pinging and setting the apps enabled value should be enough on reconnect (let them fetch the bindings on their next channel switch), but not sure.

Reply:
> Should we make a ping function for when the client gets "back online"?

@larkox I've added this function. Now the frontend reconnect handler just calls this `ping` route instead of fetching bindings. If the ping succeeds, bindings will be fetched at next channel switch.

Reply:
We decided in a meeting that we would fetch bindings during reconnect, and not use a ping route to check for presence.